mailbox.c

```diff
@@ -774,7 +774,9 @@ int mutt_mailbox_check(int force)
   struct MailboxNode *np = NULL;
   STAILQ_FOREACH(np, &AllMailboxes, entries)
   {
-    mailbox_check(np->m, &contex_sb, check_stats);
+    mailbox_check(np->m, &contex_sb, check_stats || (!np->m->first_check_stats_done && MailCheckStats));
+    if (!np->m->first_check_stats_done)
+      np->m->first_check_stats_done = true;
```

Comment:
As a matter of preference, I would just always assign no matter what the previous value is..